src/io/flutter/run/FlutterAppState.java

```diff
@@ -48,7 +49,13 @@ protected ProcessHandler startProcess() throws ExecutionException {
     Project project = getEnvironment().getProject();
     String workingDir = project.getBasePath();
     assert workingDir != null;
-    ConnectedDevice device = service.getConnectedDevices().iterator().next();
+
+    Collection<ConnectedDevice> devices = service.getConnectedDevices();
+    if (devices.isEmpty()) {
+      throw new ExecutionException("No connected device");
```

Comment:
I don't think it will be going forward, but right now the run button can be enabled w/o any devices connected.